spec/features/admin/admin_can_create_genres_spec.rb

```diff
@@ -38,5 +33,34 @@
 
       expect(page).to have_content(genre_name)
     end
+
+    context 'and I try to create an invalid genre' do
+      scenario 'I see an error message' do
+        allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@admin)
+
+        visit genres_path
+
+        click_on 'Create Genre'
+
+        expect(page).to have_content('There was an error saving that genre. Name can\'t be blank.')
+      end
+
+      scenario 'the genre input box has not changed' do
+        allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@admin)
+        # Carriage return as input char allows us to fail the Genre#save method
+        # But maintain sending the character back to the render view
+        char = "\u000D"
+        visit genres_path
+
+        within('form#new_genre') do
+          fill_in 'genre[name]', with: char
+        end
+
+        click_on 'Create Genre'
+
+        result = find('#genre_name').value
+        expect(result).to eq(char)
```

Comment:
nice addition, though it's also best practice to strip whitespace off the start/end of a user's input which would have stripped out the carriage return in this scenario :)